influxdb_client_3/__init__.py

```diff
@@ -194,10 +212,14 @@ def query(self, query, language="sql", mode="all", database=None,**kwargs ):
             database = self._database
         
         try:
-            headers = [(b"authorization", f"Bearer {self._token}".encode('utf-8'))]
-    
             # Create an authorization header
-            _options = FlightCallOptions(headers=headers, **kwargs)
+            args = {"headers": [(b"authorization", f"Bearer {self._token}".encode('utf-8'))]}
+            custom_args = {key: value for key, value in kwargs.items()}
+            # Merge defaults with user-provided arguments
+            _deep_merge(args, custom_args)
+            print(args["headers"])
+            _options = FlightCallOptions(headers=args["headers"])
```

Comment:
So we still need **kwargs:
```
_options = FlightCallOptions(headers=headers, **kwargs)
```
Since we have a user using those to add timeouts to the FlightCallOptions
https://arrow.apache.org/docs/python/generated/pyarrow.flight.FlightCallOptions.html

I wonder if this now makes more sense to define custom headers on client initialization and these are inherited by the query function. Let me know what you think

Reply:
> I wonder if this now makes more sense to define custom headers on client initialization and these are inherited by the query function. Let me know what you think

I lean toward keeping them specific to the query. I like holding close to the pyarrow Flight API--I can go there to figure it out if I need to. We could allow both and let the query override client settings, but that might be YAGNI.